### PR TITLE
Fix empty and cancelled global search

### LINKED ISSUE #1008: Global Search: "Search" should be insensitive with empty entry
## What Happened

When performing a global search, you can initiate a search with an empty search field, which highlights ever single file with hundreds of matches

## Expected Behavior

The "Search" button should be insensitive unless there is input

## Steps to Reproduce

1. Have a file open so that you can open the global search dialog
2. Search without entering any input
3. Notice that all files match hundreds of times

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #1008 

Also rebases the dialog on Granite.MessageDialog, similar to the NewBranchDialog.

### CODE REVIEW

**Comment by jeremypw** [comment]
GlobalSearchDialog

File: src/Dialogs/GlobalSearchDialog.vala [hunk 1]
```diff
@@ -51,76 +51,78 @@ public class Scratch.Dialogs.GlobalSearchDialog : Granite.Dialog {
...
-                target_val.set_boolean (src_val.get_string ().length >= 3);
-            }
+        search_term_entry.bind_property (
+            "is-valid", search_button, "sensitive", BindingFlags.DEFAULT | BindingFlags.SYNC_CREATE
         );
 
-        search_term_entry.activate.connect (() => {
-            response (search_term_entry.text != "" ? Gtk.ResponseType.ACCEPT : Gtk.ResponseType.CLOSE);
+        search_term_entry.changed.connect (() => {
+            search_term_entry.is_valid = search_term_entry.text.length >= 3;
```
Comment (danrabbit) [inline src/Dialogs/GlobalSearchDialog.vala:125]: Does it really need to be >= 3 or just 0 ?

**Review by danrabbit — CHANGES_REQUESTED** [review]
I opened #1013 with a few requested changes

File: src/Dialogs/GlobalSearchDialog.vala (hunk 1 above)
Comment (jeremypw) [reply src/Dialogs/GlobalSearchDialog.vala:112]: The rules for legal branch names are quite complicated, but single letter names are accepted by git (not sure if GitHub imposes any additional restrictions).  Not sure where I got the existing Regex and min length from but they need relaxing to exactly match git rules.  I just noticed that there is a `is_valid_name` static function provided in the git library so I'll see if I can use that.

Comment (danrabbit) [reply src/Dialogs/GlobalSearchDialog.vala:112]: @jeremypw that sounds like a good idea for the git branch, but this is the global search branch ;) Maybe some coffee or sleep? lol

**Review by danrabbit — APPROVED** [review]

**Pull request merged**